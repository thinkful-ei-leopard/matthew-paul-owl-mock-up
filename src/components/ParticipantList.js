import React from "react";
import Participant from "./Participant";

function ParticipantList(props) {
  let inSession = props.participants.filter(person => person.inSession);
  let offSession = props.participants.filter(person => !person.inSession);
  return (
    <div className="Participant-List">
      <div className="inSession">
        {inSession.map(person =>
          <Participant
            key={person.id}
            name={person.name}
            avatar={person.avatar}
            inSession={person.inSession}
            onStage={person.onStage}
          />
        )}
      </div>
      <div className="outSession">
        {offSession.map(person =>
          <Participant
            key={person.id}
            name={person.name}
            avatar={person.avatar}
            inSession={person.inSession}
            onStage={person.onStage}
          />
        )}
      </div>
    </div>
  );
}

export default ParticipantList;
