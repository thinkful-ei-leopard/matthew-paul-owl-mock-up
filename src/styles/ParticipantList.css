.Participant-List {
  width: 300px;
  padding: 10px;
  position: relative;
  text-align: left;
  overflow-y: scroll;
  border: 2px solid black;
}

.Participant {
  width: 100%;
  display: flex;
  padding: .3rem;
}

.Participant-Name {
  font-weight: 500;
}

.Participant-Avatar {
width: 40px;
border-radius: 50%;
}

.Participant-Info {
  display: flex;
  flex-direction: column;
}

.Participant-Session-Status {
  display: flex;
}

.online {
  background-color: rgb(20, 199, 20);
  width: 10px;
  height: 10px;
  border-radius: 50%;
  margin-top: 5px;
  margin-right: 5px;
}

.offline {
  background-color: rgb(121, 121, 121);
  width: 10px;
  height: 10px;
  border-radius: 50%;
  margin-top: 5px;
  margin-right: 5px;
}



