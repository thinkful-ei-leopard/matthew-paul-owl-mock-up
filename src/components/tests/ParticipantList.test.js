import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantList from '../ParticipantList';
import STORE from '../../STORE';
import renderer from 'react-test-renderer'; 

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<ParticipantList participants={STORE.participants}/>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<ParticipantList participants={STORE.participants}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });