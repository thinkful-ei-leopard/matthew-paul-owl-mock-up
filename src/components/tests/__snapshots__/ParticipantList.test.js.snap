// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Participant-List"
>
  <div
    className="inSession"
  >
    <div
      className="Participant"
    >
      <img
        alt="Avatar"
        className="Participant-Avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <span
        className="Participant-Name"
      >
        Koren Templeton
      </span>
      <div
        className="online"
      />
      <span
        className="onStage"
      />
    </div>
    <div
      className="Participant"
    >
      <img
        alt="Avatar"
        className="Participant-Avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <span
        className="Participant-Name"
      >
        Caty Flucker
      </span>
      <div
        className="online"
      />
      <span
        className="hidden"
      />
    </div>
    <div
      className="Participant"
    >
      <img
        alt="Avatar"
        className="Participant-Avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <span
        className="Participant-Name"
      >
        Frank Runciman
      </span>
      <div
        className="online"
      />
      <span
        className="hidden"
      />
    </div>
    <div
      className="Participant"
    >
      <img
        alt="Avatar"
        className="Participant-Avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <span
        className="Participant-Name"
      >
        Ashla Attwool
      </span>
      <div
        className="online"
      />
      <span
        className="onStage"
      />
    </div>
  </div>
  <div
    className="outSession"
  >
    <div
      className="Participant"
    >
      <img
        alt="Avatar"
        className="Participant-Avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <span
        className="Participant-Name"
      >
        Axe Kubicka
      </span>
      <div
        className="offline"
      />
      <span
        className="hidden"
      />
    </div>
  </div>
</div>
`;
